nmpc_controller/src/quad_nlp.cpp

```diff
@@ -1392,14 +1396,24 @@ void quadNLP::update_solver(
     lambda0_.fill(1000);
 
     // Initialize current state
-    get_primal_state_var(w0_, 0) = x_current_;
+    get_primal_body_state_var(w0_, 0) = x_current_.head(n_body_);
+    if (n_vec_[0] > config_.x_dim_simple) {
+      get_primal_foot_state_var(w0_, 0).head(n_foot_ / 2) =
+          foot_pos_world_.row(0);
+      get_primal_foot_state_var(w0_, 0).tail(n_foot_ / 2) =
+          foot_vel_world_.row(0);
+      get_primal_joint_state_var(w0_, 0) = x_current_.tail(n_joints_);
+    }
 
     // Initialize future state predictions and controls
     for (size_t i = 0; i < N_ - 1; i++) {
-      get_primal_state_var(w0_, i + 1).head(n_simple_) =
-          x_reference_.col(i + 1).head(n_simple_);
-      if (n_vec_[i + 1] == n_complex_) {
-        get_primal_state_var(w0_, i + 1).tail(n_null_) = x_null_nom_;
+      get_primal_body_state_var(w0_, i + 1) = x_reference_.col(i + 1);
+      if (n_vec_[i + 1] == config_.x_dim_complex) {
+        get_primal_foot_state_var(w0_, i + 1).head(n_foot_ / 2) =
+            foot_pos_world_.row(i + 1);
+        get_primal_foot_state_var(w0_, i + 1).tail(n_foot_ / 2) =
```

Comment:
Is this in world or body coordinates? I remember that in my casadi setup, the vector was p_lb, meaning from foot to body.

Reply:
Yes, but you didn't update foot_pos_body_

Reply:
It gets updated in NMPCController::computeLegPlan (not the cleanest argument passing, I suppose...)